### PR TITLE
ddl: use a global owner manager instance for DDL, to avoid owner change

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #57185

Problem Summary:

### What changed and how does it work?
- use a global owner manager instance for DDL, to avoid owner change during bootstrap, we want it stable. or change back to older version node after upgrade, we want the owner stick on the newer version node.
- fix br owner manager not closed
- fix br/bindinfo owner manager report 'the client connection is closing' as the campaign loop exit after client close

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

bootstrap with 3 TiDB of this PR, success, and the owner on first Domain start is still the owner after second start.

upgrade from a 3 TiDB cluster of version v8.1.1, success, and after upgrade, the newer version node is still the owner

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/domain.go [hunk 1]
```diff
@@ -1311,62 +1313,6 @@ func NewDomain(store kv.Storage, schemaLease time.Duration, statsLease time.Dura
 
 const serverIDForStandalone = 1 // serverID for standalone deployment.
 
-// NewEtcdCli creates a new clientv3.Client from store if the store support it.
```
Comment (D3Hunter) [inline pkg/domain/domain.go:1314]: moved to `store` pkg to avoid import cycle

**Review by wjhuang2016 — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by yudongusa — APPROVED** [review]

File: pkg/ddl/owner_mgr.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	id       string
+	ownerMgr owner.Manager
+	started  bool
+}
+
+// Start starts the TiDBInstance.
+func (om *ownerManager) Start(ctx context.Context, store kv.Storage) error {
+	// BR might start domain multiple times, we need to avoid it. when BR have refactored
+	// this part, we can remove this.
+	if om.started {
```
Comment (D3Hunter) [inline pkg/ddl/owner_mgr.go:57]: cc @Leavrth , please help remove this part after BR refactor

**Review by Leavrth — APPROVED** [review]

**Pull request merged**